File: java/2525.categorize-box-according-to-criteria.java
/*
 * @lc app=leetcode.cn id=2525 lang=java
 * 
 * @lcpr version=30102
 *
 * [2525] 根据规则将箱子分类
 */


// @lcpr-template-start
import java.util.ArrayList;
import java.util.List;
import java.util.Collections;
import java.util.Map;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Arrays;
import java.util.Set;
import java.util.ArrayDeque;
import java.util.Deque;

// @lcpr-template-end
// @lc code=start
class Solution {
	public String categorizeBox(int length, int width, int height, int mass) {
		boolean isHeavy = mass >= 100;
		boolean isBulky = (long) length * width * height >= 1_000_000_000
				|| Math.max(length, Math.max(width, height)) >= 10000;
		if (isBulky && isHeavy)
			return "Both";
		else if (isBulky)
			return "Bulky";
		else if (isHeavy)
			return "Heavy";
		else
			return "Neither";
	}
}
// @lc code=end



/*
// @lcpr case=start
// 1000\n35\n700\n300\n
// @lcpr case=end

// @lcpr case=start
// 200\n50\n800\n50\n
// @lcpr case=end

 */

File: golang/2525.categorize-box-according-to-criteria.go
/*
 * @lc app=leetcode.cn id=2525 lang=golang
 * @lcpr version=30102
 *
 * [2525] 根据规则将箱子分类
 */

// @lcpr-template-start
package leetcode_go

// @lcpr-template-end
// @lc code=start
func categorizeBox(length int, width int, height int, mass int) string {
	MAX_NUM := 10000
	MAX_V := 1000000000
	v := length * width * height
	set := make(map[string]bool)
	if length >= MAX_NUM || width >= MAX_NUM || height >= MAX_NUM || mass >= MAX_NUM || v >= MAX_V {
		set["Bulky"] = true
	}
	if mass >= 100 {
		set["Heavy"] = true
	}
	if set["Heavy"] && set["Bulky"] {
		return "Both"
	} else if set["Heavy"] && !set["Bulky"] {
		return "Heavy"
	} else if set["Bulky"] && !set["Heavy"] {
		return "Bulky"
	} else {
		return "Neither"
	}
}

// @lc code=end

/*
// @lcpr case=start
// 1000\n35\n700\n300\n
// @lcpr case=end

// @lcpr case=start
// 200\n50\n800\n50\n
// @lcpr case=end

*/
